src/main.ts

```diff
@@ -40,11 +44,29 @@ export async function runSorald(
     'stats.json'
   );
 
+  const changedLines =
+    ratchetFrom === undefined
```

Comment:
Since you specified in the README that `ratchetFrom` usually equals `'origin/main'` or `'origin/master'`, you can assign it a default value. I know there are two values to choose from but I think `git` will have a utility to get the default branch.

Reply:
The default is to not ratchet, i.e. to process all rule violations, regardless of it being in changed code or not. Thus, the default value is undefined.